lib/puppet/provider/oneview_volume/c7000.rb

```diff
@@ -27,9 +27,30 @@
 
   # Provider methods
   def exists?
+    create
     super([nil, :found, :get_attachable_volumes, :get_extra_managed_volume_paths])
   end
 
+  def create
+    set_template_uri if resource['data'].key?('templateUri')
+    @data = resource['data']
+    return true if resource_update
+    super(:create)
+  end
+
+  def set_template_uri
+    volume_template_class = OneviewSDK.resource_named('VolumeTemplate', @client.api_version)
+    resource['data']['templateUri'] = volume_template_class.get_all(@client).first['uri'] unless resource['data']['templateUri']
+    set_storage_pool
+  end
+
+  def set_storage_pool
+    storage_pool_class = OneviewSDK.resource_named('StoragePool', @client.api_version)
+    snap_uri = storage_pool_class.find_by(@client, name: 'CPG_FC-AO').first['uri']
```

Comment:
storage pool name is hard coded here. What if user wants to use different pool..? please avoid hardcoded values

Reply:
Done